### PR TITLE
Free disk space before running GoReleaser

### PR DESCRIPTION
**Pull request by rquitales**
Note: this PR will be replicated in ci-mgmt as it's our SoT for workflows. It is proposed here to provide a shorter feedback loop for testing purposes.

### Related issues (optional)

Fixes: #2874

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ on:
     - master
     - main
     - feature-**
+    - rquitales/fix-disk-space-ci
```
Comment (rquitales) [inline .github/workflows/build.yml:10]: To test that this solution works before merging to `master`. Will be removed on submit.

**Comment by rquitales** [comment]
Build action running on test branch to demonstrate that this fix works: https://github.com/pulumi/pulumi-kubernetes/actions/runs/8257324298

**Review by mjeffryes — APPROVED** [review]
Note that this change is already in ci-mgmt from https://github.com/pulumi/ci-mgmt/pull/845

**Comment by rquitales** [comment]
Ah, closing this in favor of #2586 in that case.

**Pull request closed without merging**